pkg/snowflake/builder.go

```diff
@@ -0,0 +1,419 @@
+package snowflake
+
+import (
+	"database/sql"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+type ParamType string
+
+var (
+	Integer    ParamType = "int"
+	String     ParamType = "string"
+	StringList ParamType = "stringlist"
+)
+
+type SQLParameter struct {
+	name      string
+	paramType ParamType
+}
+
+type BuilderConfig struct {
+	beforeObjectType map[string]string
+	afterObjectType  map[string]string
+	parameters       map[string]*SQLParameter
```

Comment:
could this be `[]*SQLParameter`? If you use the name as a field on the struct it should be okay. Benefit is that it preserves order.